src/mappings/gvault.ts

```diff
@@ -1,15 +1,73 @@
 import {
- LogStrategyHarvestReport
+ LogStrategyHarvestReport,
+ LogWithdrawalFromStrategy,
+ LogStrategyAdded
 } from '../../generated/GVault/GVault';
 import {
 	updateGTokenFactor as updateFactor
 } from '../setters/factors';
 import { setGvtPrice } from '../setters/price';
+import { 
+  setGVaultHarvest,
+  setStrategyQueue,
+  setGVaultStrategy,
+  setStrategyWithdraw
+} from '../setters/strats';
+import { tokenToDecimal } from '../utils/tokens';
+import { DECIMALS, NUM } from '../utils/constants';
+import { Address } from '@graphprotocol/graph-ts';
 
 export function handleStrategyHarvestReport(event: LogStrategyHarvestReport): void {
+    //Save HarvestReport
+    const harvest = setGVaultHarvest(
+      event.transaction.hash.toHex() + "-" + event.logIndex.toString(),
+      event.params.strategy.toHexString(),
+      tokenToDecimal(event.params.gain, 18, DECIMALS),
+      tokenToDecimal(event.params.loss, 18, DECIMALS),
+      tokenToDecimal(event.params.debtPaid, 18, DECIMALS),
+      tokenToDecimal(event.params.debtAdded, 18, DECIMALS),
+      tokenToDecimal(event.params.lockedProfit, 18, DECIMALS),
+      event.block.timestamp,
+    );
+
+    // update GVault strategy
+    setGVaultStrategy(
+      event.params.strategy,
+      'harvest',
+      harvest.debtPaid,
+      harvest.debtAdded,
+      NUM.ZERO,
+      event.block.number
+    )
+
     // update factor
     updateFactor(event.block.timestamp.toI32());
 
     // update gvt price
     setGvtPrice();
-  }
\ No newline at end of file
+}
+
+export function handleWithdrawalFromStrategy(event: LogWithdrawalFromStrategy): void {
+    const withdraw = setStrategyWithdraw(
+      event.transaction.hash.toHex() + "-" + event.logIndex.toString(),
+      event.params.strategyId,
+      tokenToDecimal(event.params.strategyDebt, 18, DECIMALS),
+      tokenToDecimal(event.params.totalVaultDebt, 18, DECIMALS),
+      tokenToDecimal(event.params.lossFromStrategyWithdrawal, 18, DECIMALS),
+      event.block.timestamp
+    )
+
+    // update GVault strategy
+    setGVaultStrategy(
+      Address.fromString(withdraw.strategyAddress),
+      'withdraw',
+      NUM.ZERO,
+      NUM.ZERO,
+      withdraw.strategyDebt,
+      event.block.number
+    )
+}
+
+export function handleStrategyAdded(event: LogStrategyAdded): void {
+  setStrategyQueue(event.params.strategy, event.params.id.toI32().toString())
+}
```

Comment:
If we have to create a new strategy, either way we will have to update file `src->utils->strats.ts`. Therefore, I would remove this handler and entity `StrategyQueue` and add field `stratId` or `queueId` or however we want to name it in the corresponding strategy.

Reply:
if we put the strategyId to strategy, which should be the ID? strategyAddress or strategyId. for Harvest event, the event has strategyAddress, but for withdraw the event has strategyId. no matter which field to be ID. we need use other to find the strategy?

Reply:
I suggest to add field `queueId` into every strategy at `src->utils->strats.ts`. I would keep the strategy address as the main ID, and queueId just as a field in the strategy.
If we need to know the strategy address by the queue id, we need a simple function to provide this value, as the ones defined in the same `src->utils->strats.ts`, such as `getAdapterAddressByStrategy()`, `getAdaptersByCoin()`, etc.  In this case, we would have `getStrategyAddressByQueue()`